### PR TITLE
feat: enable 20MB request limit, this feature is allowlist only.

### PR DESCRIPTION
**Pull request by yirutang**
Thank you for opening a Pull Request! Before submitting your PR, there are a few things you can do to make sure it goes smoothly:
- [ ] Make sure to open an issue as a [bug/issue](https://github.com/googleapis/java-bigquerystorage/issues/new/choose) before writing your code!  That way we can discuss the change, evaluate designs, and agree on the general idea
- [ ] Ensure the tests and linter pass
- [ ] Code coverage does not decrease (if any source code was changed)
- [ ] Appropriate docs were updated (if necessary)

Fixes #<issue_number_goes_here> ☕️

If you write sample code, please follow the [samples format](
https://github.com/GoogleCloudPlatform/java-docs-samples/blob/main/SAMPLE_FORMAT.md).

### CODE REVIEW

**Review by GaoleMeng — APPROVED** [review]

File: google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1/StreamWriter.java [hunk 1]
```diff
@@ -771,6 +774,16 @@ public Builder setRetrySettings(RetrySettings retrySettings) {
       return this;
     }
 
+    /**
+     * Enables 20MB request size.
+     * Project has to be explicitly enabled for the feature to work.
+     * @return
+     */
+    public Builder enableLargerRequest() {
```
Comment (GaoleMeng) [inline google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1/StreamWriter.java:782]: only on streamWriter?

Comment (yirutang) [reply google-cloud-bigquerystorage/src/main/java/com/google/cloud/bigquery/storage/v1/StreamWriter.java:782]: Yes, and I am still writing a e2e test.

File: google-cloud-bigquerystorage/src/test/java/com/google/cloud/bigquery/storage/v1/it/ITBigQueryWriteManualClientTest.java [hunk 2]
```diff
@@ -1541,4 +1543,100 @@ public void testMultiplexingMixedLocation()
...
+        int size = sizeSet.get(new Random().nextInt(3));
+        LOG.info("Sending request of size: " + size);
+        responseList.add(
+            streamWriter.append(
+                CreateProtoRows(new String[] {new String(new char[size]).replace("\0", "a")})));
+      }
+      for (int i = 0; i < 10; i++) {
+        assertFalse(responseList.get(i).get().hasError());
+      }
+      TableResult queryResult =
```
Comment (GaoleMeng) [inline google-cloud-bigquerystorage/src/test/java/com/google/cloud/bigquery/storage/v1/it/ITBigQueryWriteManualClientTest.java:1587]: can we assert on the stored string length

File: google-cloud-bigquerystorage/src/test/java/com/google/cloud/bigquery/storage/v1/it/ITBigQueryWriteManualClientTest.java [hunk 3]
```diff
@@ -1541,4 +1543,100 @@ public void testMultiplexingMixedLocation()
...
+            .build()) {
+      List<Integer> sizeSet = Arrays.asList(200, 10 * 1024 * 1024, 19 * 1024 * 1024);
+      List<ApiFuture<AppendRowsResponse>> responseList =
+          new ArrayList<ApiFuture<AppendRowsResponse>>();
+      for (int i = 0; i < 10; i++) {
+        int size = sizeSet.get(new Random().nextInt(3));
+        LOG.info("Sending request of size: " + size);
+        responseList.add(
+            streamWriter.append(
+                CreateProtoRows(new String[] {new String(new char[size]).replace("\0", "a")})));
```
Comment (GaoleMeng) [inline google-cloud-bigquerystorage/src/test/java/com/google/cloud/bigquery/storage/v1/it/ITBigQueryWriteManualClientTest.java:1582]: can we use a more random string. We had experience on some compression will work along the way, we can eliminate that effect by random

File: google-cloud-bigquerystorage/src/test/java/com/google/cloud/bigquery/storage/v1/it/ITBigQueryWriteManualClientTest.java (hunk 2 above)
Comment (yirutang) [reply google-cloud-bigquerystorage/src/test/java/com/google/cloud/bigquery/storage/v1/it/ITBigQueryWriteManualClientTest.java:1592]: The query result may not be able to return 20MB back.

**Pull request merged**